examples/adiabatic-3SAT/README.md

```diff
@@ -0,0 +1,83 @@
+# Adiabatic evolution for solving an Exact Cover problem
+
+Code at: [https://github.com/Quantum-TII/qibo/tree/adiabatic-3SAT/examples/adiabatic-3SAT](https://github.com/Quantum-TII/qibo/tree/adiabatic-3SAT/examples/adiabatic-3SAT)
+
+## Introduction
+
+Adiabatic quantum computation aims to reach the ground state of a problem Hamiltonian encoding the solution of a hard problem, by adiabatically evolving a system from the ground state of a known, easy to prepare, Hamiltonian to the problem Hamiltonian.
+
+An Exact Cover instance of a 3SAT problem is characterized by a set of clauses containing 3 bits that are considered satisfied if one of them is in position 1, while the other remain at 0. The solution of this instance is bitstring that fulfills all the clauses at the same time.
+
+## Adiabatic evolution
+
+Adiabatic compution deals with finding the ground state of a Hamiltonian that encodes the solution of a computational problem. This has to be mapped to the time dependent Hamiltonian in the form
+
+![hamiltonian](images/H.png)
+
+where H_0 and H_p are the initial and problem Hamiltonian respectively, and s(t) defines the schedule that the adiabatic evolution follows. How fast this evolution can be performed depends on the minimum gap energy of the system during the evolution, that is the difference of energy between the ground state and the first excited state along the schedule. The smaller the gap energy the slower the evolution has to be.
+
+The initial Hamiltonian used in this example is the following, 
+
+![initial-hamiltonian](images/h0.png)
+
+which has a ground state of the equal superposition of all possible computational states.
+
+As for the problem Hamiltonian that encodes the solution of an Exact cover problem, first we define the operator
+
+![z-operator](images/z.png)
```

Comment:
This image appears broken. I think it should be `z-matrix.png` instead of `z.png`.